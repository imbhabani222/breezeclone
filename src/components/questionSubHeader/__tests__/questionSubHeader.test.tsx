import React from "react";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import QuestionHeader, { SubHeaderData } from "../QuestionSubHeader";
import listIcon from "../../../assets/list_item.svg";

describe("QuestionHeader Testing", () => {
  const handleClick = jest.fn();
  const defaultProps: SubHeaderData = {
    typeOfQuestions: "Fill In the Blanks",
    Icon: "list_item.svg",
    title: "Answer",
    text: "Please choose correct answer.",
    buttons: [
      {
        type: "rectFilled",
        display: "REPORT AN ISSUE",
        onClick: handleClick,
        beforeIcon: "beforeIcon",
      },
    ],
  };

  function renderQuestionHeader(props: SubHeaderData) {
    console.log({ ...defaultProps, ...props });
    return render(<QuestionHeader {...defaultProps} titles={props} />);
  }

  afterEach(cleanup);
  test("Render without Crashing", () => {
    renderQuestionHeader(defaultProps);
  });
  test("Match snapshot questionSubHeader component", () => {
    const { asFragment } = renderQuestionHeader(defaultProps);
    expect(asFragment()).toMatchSnapshot();
  });
  test("renders the icon src and alt name", () => {
    renderQuestionHeader({
      ...defaultProps,
      text: "Please choose correct question.",
    });
    console.log("change");
    const image = screen.getByAltText("Fill In the Blanks");
    expect(image).toHaveAttribute("src", "list_item.svg");
  });
  test("text is null", () => {
    renderQuestionHeader({
      ...defaultProps,
      text: "",
    });
    console.log("test");
  });
});

// test("click on button icon", async () => {
//   renderQuestionHeader(mock.buttons);
//   fireEvent(
//     screen.getByTestId("button-up"),
//     new MouseEvent("click", {
//       bubbles: true,
//       cancelable: true,
//     })
//   );
// });
