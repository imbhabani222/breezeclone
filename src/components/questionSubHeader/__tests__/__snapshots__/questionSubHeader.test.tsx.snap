// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionHeader Testing Match snapshot questionSubHeader component 1`] = `
<DocumentFragment>
  <div
    class="question_type"
  >
    <div
      class="image"
    >
      <img
        alt="Fill In the Blanks"
        src="list_item.svg"
      />
      <h3>
        Fill In the Blanks
      </h3>
    </div>
    <div>
      <div
        class="btnGrp header_btn_div"
      >
        <button
          class="ant-btn ant-btn-rectFilled btn rectFilledBtn"
          data-testid="button-up-REPORT AN ISSUE"
          type="button"
        >
          <img
            alt="beforeIcon"
            src="beforeIcon"
          />
          <span>
            REPORT AN ISSUE
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="question_header"
  >
    <div
      class="header_content"
    >
      <div
        class="ant-typography content_title"
      >
        Answer
      </div>
      <div
        class="ant-typography content_text"
      >
        Please choose correct answer.
      </div>
    </div>
  </div>
</DocumentFragment>
`;
